backend/controllers/analytics.js

```diff
@@ -65,3 +69,50 @@ module.exports.application = (req, res) => {
             });
         });
 };
+
+module.exports.focusItem = async (req, res) => {
+    var students = [];
+    if(req.user.teacher_id == undefined){
+        return res.status(401).json({
+            status: 'error',
+            message: "You don't have permission for this"
+        });
+    }
+    var analytics = {};
+    if(req.body.student != undefined || req.body.students != undefined){
+        if(req.body.student != undefined){
+            students.push(mongoose.Types.ObjectId(req.body.student));
+        }else{
+            students = req.body.students.map(mongoose.Types.ObjectId);
+        }
+        for(index in students){
```

Comment:
Just fyi, you can do `for (student of students)` which will give you the element in the array rather than the index. Not worth it to change here.

Reply:
I google a decent bit to try and find how to do that. I hate using `in` for for each loops, so I might change that anyhow.